src/vm-vcpu/src/vm.rs

```diff
@@ -27,19 +28,26 @@ use vm_vcpu_ref::aarch64::interrupts::{self, Gic, GicConfig, GicState};
 #[cfg(target_arch = "x86_64")]
 use vm_vcpu_ref::x86_64::mptable::{self, MpTable};
 
+#[cfg(target_arch = "aarch64")]
+pub const MAX_IRQ: u32 = interrupts::MIN_NR_IRQS;
+#[cfg(target_arch="x86_64")]
+pub const MAX_IRQ:u8 = mptable::IRQ_MAX;
```

Comment:
Let's define this as u32 as well so that we can keep the code simpler for both platforms. If we want to have one as u32 and one as u8, then this type difference should be passed to the `VmConfig` structure as well, and that would needlessly complicate the code. Instead, we can define both as u32.

On x86, for now, we need the IRQ to be an u8. We can do a `try_from` when we initialize the `MpTable`, and return an error if the number is too big. I am saying for now because in the future we might choose to switch to I/O APIC and thus increase the number of available IRQs.